### PR TITLE
feat: POC

### PR DESCRIPTION
**Pull request by roeeyud**

### CODE REVIEW

File: src/context/Controller.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export const Context = createContext();
+export function Provider({ children }) {
+    const [values, setValues] = useState({});
+
+    function sendNewState(newState) {
+        var request = new XMLHttpRequest();
+            
+        request.open('POST', '/values', newState);
+        request.send();
```
Comment (LiorRabinovich) [inline src/context/Controller.js]: Why not use axois or fetch?
What is POST to `/values`?

File: src/context/Controller.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    function onControllerChange(name, value) {
+        const newState = {
+            ...values,
+            [name]: value,
+        };
+        sendNewState(newState);
+        setValues(newState);
```
Comment (LiorRabinovich) [inline src/context/Controller.js]: Do you think it would be a better indicator if you print to console?

File: src/components/GamepadComponent.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import Gamepad from '@material-ui/icons/Gamepad';
+
+import { Context as ControllerContext } from '../context/Controller';
+
+export default function GamepadComponent() {
+    const { onControllerChange } = useContext(ControllerContext);
+
+    return <Fab aria-label="peripheralController" 
+        onMouseDown={() => onControllerChange('demo-value', 'down')}
+        onMouseUp={() => onControllerChange('demo-value', 'up')}
```
Comment (LiorRabinovich) [inline src/components/GamepadComponent.js]: Why not four buttons: right, left, up, down?

File: src/App.js [hunk 4]
```diff
@@ -2,14 +2,19 @@ import React from 'react';
...
 export default function App() {
   return (
     <Router>
-    <div className="container">
-      <Route path="/" exact component={Home} />
-      <Route path="/page" component={Page} />
-    </div>
+      <ControllerProvider>
+        <div className="container">
+          <GamepadComponent />
```
Comment (LiorRabinovich) [inline src/App.js]: What do you think about putting GamepadComponent into one of the routes and that way we will also have routes of more stuff and not just a game pad

Comment (roeeyud) [reply src/App.js]: I don't see the need for routes at this point###

File: src/components/GamepadComponent.js (hunk 3 above)
Comment (roeeyud) [reply src/components/GamepadComponent.js]: This is only an example,

File: src/context/Controller.js (hunk 2 above)
Comment (roeeyud) [reply src/context/Controller.js]: done

File: src/components/GamepadController.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            setConnected(true); 
+        });
+        window.addEventListener('gamepaddisconnected', function(e) {
+            console.log('gamepadconnected');
+            setConnected(false);
+        });
+
+        return function cleanup() {
+            
+        };
```
Comment (LiorRabinovich) [inline src/components/GamepadController.js]: Why is there an empty cleanup function here?
Aren't you supposed to remove events when unmounted?

File: src/components/GamepadController.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    }, []);
```
Comment (LiorRabinovich) [inline src/components/GamepadController.js]: I think you can write this function in custom hooks

File: src/context/Controller.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+export const Context = createContext();
+export function Provider({ children }) {
+    const [values, setValues] = useState({});
+
+    function onControllerChange(name, value) {
+        const newState = {
+            ...values,
+            [name]: value,
+        };
```
Comment (LiorRabinovich) [inline src/context/Controller.js]: Where is the starting state?
I want some schema that can work without guessing what state parameters can get.
I'm not sure about touchX and touchY vars and I don't think we should keep state.
what do you think of just working simple without state? I don't have to keep the robot's location, it's a source of bugs.

File: src/components/TouchpadController.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    leftButton: {
+        marginRight: margin,
+    },
+    rightButton: {
+        marginLeft: margin,
+    },
+  });
+
+export default function TouchpadController() {
+    const classes = useStyles();
```
Comment (LiorRabinovich) [inline src/components/TouchpadController.js]: What do you think about doing destructuring?

File: src/components/TouchpadController.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
+import React, { useContext, useEffect, useState } from 'react';
+
+import { makeStyles } from '@material-ui/core/styles';
+import Fab from '@material-ui/core/Fab';
+import KeyboardArrowDown from '@material-ui/icons/KeyboardArrowDown';
+import KeyboardArrowUp from '@material-ui/icons/KeyboardArrowUp';
+import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
+import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
```
Comment (LiorRabinovich) [inline src/components/TouchpadController.js]: Why not `import { KeyboardArrowDown, KeyboardArrowUp, KeyboardArrowLeft, KeyboardArrowRight  } from '@material-ui/icons';`?

File: src/components/GamepadController.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext, useEffect, useState } from 'react';
+
+import Fab from '@material-ui/core/Fab';
+import Gamepad from '@material-ui/icons/Gamepad';
```
Comment (LiorRabinovich) [inline src/components/GamepadController.js]: Why not import { Fab, Gamepad } from '@material-ui/icons';?

Comment (roeeyud) [reply src/components/GamepadController.js]: That would create a much bigger client payload this way you only add used icons to the final product

File: src/components/TouchpadController.js (hunk 9 above)
Comment (roeeyud) [reply src/components/TouchpadController.js]: see icon comment

File: src/components/TouchpadController.js (hunk 8 above)
Comment (roeeyud) [reply src/components/TouchpadController.js]: that's not how it's documented...

File: src/context/Controller.js (hunk 7 above)
Comment (roeeyud) [reply src/context/Controller.js]: Leave it for now, I'll wait to hear from hardware how they want comms to work

File: src/components/GamepadController.js (hunk 6 above)
Comment (roeeyud) [reply src/components/GamepadController.js]: will do later when finalizing gamepad controller

File: src/components/GamepadController.js (hunk 5 above)
Comment (roeeyud) [reply src/components/GamepadController.js]: will do later when finalizing gamepad controller

Comment (LiorRabinovich) [reply src/components/GamepadController.js]: Why not in this PR?

File: src/components/GamepadController.js (hunk 6 above)
Comment (LiorRabinovich) [reply src/components/GamepadController.js]: Why not in this PR?

Comment (roeeyud) [reply src/components/GamepadController.js]: on it!

File: src/components/GamepadController.js (hunk 5 above)
Comment (roeeyud) [reply src/components/GamepadController.js]: On it!

Comment (roeeyud) [reply src/components/GamepadController.js]: Done.

**Pull request merged**